lib/finch.ex

```diff
@@ -147,10 +149,24 @@ defmodule Finch do
   end
 
   defp valid_opts_to_map(valid) do
+    # We need to enable keepalive and set the nodelay flag to true by default.
+    transport_opts =
+      valid
+      |> get_in([:conn_opts, :transport_opts])
+      |> List.wrap()
+      |> Keyword.put_new(:timeout, @default_connect_timeout)
+      |> Keyword.put_new(:nodelay, true)
+      |> Keyword.put(:keepalive, true)
```

Comment:
Any reason not to use `put_new` for this one too? I can't imagine many people would want to disable it, but why not make it possible like the other options?

Reply:
I don't think we want to let people opt-out of this since it means that connections are less likely to be closed correctly.